core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -685,7 +701,10 @@ object IO {
   /**
    * Sleeps for the specified duration. This is always asynchronous.
    */
-  final def sleep(duration: Duration): IO[Nothing, Unit] = new Sleep(duration)
+  final def sleep(duration: Duration): IO[Nothing, Unit] = duration match {
```

Comment:
Don't solve the problem here, though, solve it in the RTS. This should just create `new Sleep(duration)`.